File: C++/sum-of-left-leaves.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        int result = 0;
        if (root) {            
            if (root->left) {
                if (root->left->left == nullptr && root->left->right == nullptr) {
                    result += root->left->val;
                } else {
                    result += sumOfLeftLeaves(root->left);
                }
            }            
            result += sumOfLeftLeaves(root->right);
        }
        return result;
    }
};

File: Go/sum-of-left-leaves.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sumOfLeftLeaves(root *TreeNode) int {
    if root == nil {
        return 0
    }
    sum := sumOfLeftLeaves(root.Right)
    if root.Left != nil && root.Left.Left == nil && root.Left.Right == nil {
        sum += root.Left.Val
    } else {
        sum += sumOfLeftLeaves(root.Left)
    }
    return sum
}